curator-x-async/src/test/java/org/apache/curator/x/async/modeled/TestZPath.java

```diff
@@ -57,11 +57,11 @@ public void testBasic()
         assertFalse(path.startsWith(ZPath.root.child("two")));
 
         ZPath checkIdLike = ZPath.parse("/one/{two}/three");
```

Comment:
I see.  That seemed like a bug to me, but I can see now when considering the two methods the difference in contracts.  

Perhaps additional documentation would be helpful here (which I'm happy to help with).

In the parseWithId's method, I see a "TODO" comment without any explanation of what remains to be done.  Is this something we should address as well?

Do you concur that the original problem of having something like "/root" as the first parameter should be considered valid?

Reply:
> Perhaps additional documentation would be helpful here (which I'm happy to help with).

We'd really appreciate that!

> n the parseWithId's method, I see a "TODO"

Man - I don't remember :( the code looks good to me now though

> Do you concur that the original problem of having something like "/root" as the first parameter should be considered valid?

I'm not sure. Curator (and ZooKeeper itself) has historically had a lot of trouble with leading slashes and the root path (`/`). I'd like to see copious testing on this if we were to make a change. For example, over the years we've gotten PRs trying to change how `ZKPaths.java` works and they all introduce problems. I'm not saying the way it's implemented is ideal but that it may be too late on this.

Reply:
Thank you for the back history.

That does make me nervous to introduce this type of change.  And, given there are other ways to construct the path the way the I want, it may not be worth the risk.

I'd still be happy to make a few additions to the documentation (javadoc as well as those unit test cases) so that confusion for people like me is reduced in the future :)

Would it be more appropriate to re-purpose this PR, or create a new one?

Reply:
Up to you - whatever is easiest

Reply:
Sorry for the churn, but now that I'm looking at this again more closely, the current implementation does still seem to have a bug to me.

I understand that the parse() method doesn't resolve parameters, and that parseWithID's does resolve parameters.

The unit test calls parse() on a path when parameters in it.

Example:
`ZPath checkIdLike = ZPath.parse("/one/{two}/three");`

As expected, that doesn't touch the parameters, and returns a ZPath in checkIdLike of:
`/one/{two}/three`

Next, the "isResolved()" method is called on that path, and the unit test is currently expecting it to be resolved:
assertTrue(checkIdLike.isResolved());

The confusion for me is:  How is that considered a resolved path if `checkIdLike` still contains the `{two}` parameter?

My change had fixed this, so that if any parameters (items in `{}` delimiters} is found in the ZPath, it's considered unresolved.

Reply:
`ZPath.parse()` doesn't consider IDs. The use-case here people who put things in their path that look like ZPath IDs. They don't want them considered as IDs and want them treated as normal paths. So, yes, it's correct in this test that isResolved() returns true.

Reply:
I will update the documentation to indicate that resolved state is set by by the appropriate parse methods and is not indicative of the presence of parameter looking things in the path.